tests/handlers/aws/test_logstash_output.py

```diff
@@ -0,0 +1,126 @@
+        # NOTE: plain curly brackets must be escaped in this string (double them)
+        logstash_config = f"""\
+            input {{
+              http {{
+                port => {self.logstash_http_port}
+                codec => json_lines
+              }}
+            }}
+
+            output {{ stdout {{ codec => json_lines }} }}
+            """
+        lgc.with_env("CONFIG_STRING", logstash_config)
+        self.logstash = lgc.start()
+
+        self.fixtures = {
+            "cw_log_1": _load_file_fixture("cloudwatch-log-1.json"),
+            "cw_log_2": _load_file_fixture("cloudwatch-log-2.json"),
+        }
+
+        group_name = _class_based_id(self, suffix="source-group")
+        stream_name = _class_based_id(self, suffix="source-stream")
+
+        _logs_create_cloudwatch_logs_group(self.logs_client, group_name=group_name)
+        g = _logs_create_cloudwatch_logs_stream(self.logs_client, group_name=group_name, stream_name=stream_name)
+        cloudwatch_group_arn = g["arn"]
+
+        _logs_upload_event_to_cloudwatch_logs(
+            self.logs_client,
+            group_name=group_name,
+            stream_name=stream_name,
+            messages_body=[self.fixtures["cw_log_1"] + self.fixtures["cw_log_2"]],
+        )
+
+        self.group_name = group_name
+        self.stream_name = stream_name
+
+        config_content = _load_file_fixture("config.yaml")
+        self.config = Template(config_content).substitute(
+            dict(CloudwatchLogStreamARN=cloudwatch_group_arn, LogstashURL=self.logstash.get_url())
+        )
+
+        config_bucket_name = _class_based_id(self, suffix="config-bucket").lower()
+        config_file_path = "folder/config.yaml"
+        _s3_upload_content_to_bucket(
+            client=self.s3_client,
+            content=self.config,
+            content_type="text/plain",
+            bucket_name=config_bucket_name,
+            key=config_file_path,
+        )
+
+        os.environ["AWS_DEFAULT_REGION"] = aws_default_region
+        os.environ["S3_CONFIG_FILE"] = f"s3://{config_bucket_name}/{config_file_path}"
+        os.environ["SQS_CONTINUE_URL"] = _sqs_create_queue(self.sqs_client, _class_based_id(self, suffix="-continuing"))
+        os.environ["SQS_REPLAY_URL"] = _sqs_create_queue(self.sqs_client, _class_based_id(self, suffix="-replay"))
+
+        self.mocks = {
+            "s3client": mock.patch("storage.S3Storage._s3_client", new=self.s3_client),
+            "cloudwatchclient": mock.patch("handlers.aws.utils.get_cloudwatch_logs_client", lambda: self.logs_client),
+            "sqsclient": mock.patch("handlers.aws.utils.get_sqs_client", lambda: self.sqs_client),
+        }
+        for k, m in self.mocks.items():
+            m.start()
+
+    def tearDown(self) -> None:
+        self.localstack.stop()
+        self.logstash.stop()
+
+        for k, m in self.mocks.items():
+            m.stop()
+
+    def test_foo(self) -> None:
```

Comment:
Maybe we could finally give a name to this test, now that everything is so tidy :)